templates/_helpers.tpl

```diff
@@ -145,16 +145,16 @@
 {{/* --------------- START OF TEMPLATE ------------- */}}
 
 {{- define "gitservers" -}}
-{{- include "joinFmt" (dict "count" (int (default 1 .Values.cluster.gitserver.shards)) "fmt" "gitserver-%d:3178" "sep" " ") -}}
+{{- include "joinFmt" (dict "count" (int (default 1 .Values.cluster.gitserver.shards)) "fmt" "gitserver-%d.gitserver.default.svc.cluster.local:3178" "sep" " ") -}}
```

Comment:
Before, this function returned 

```
"gitserver-1:3178 gitserver-2:3178 gitserver-3:3178 ..."
```

it now returns: 
```
"gitserver-0.gitserver.default.svc.cluster.local:3178 gitserver-1.gitserver.default.svc.cluster.local:3178 gitserver-2.gitserver.default.svc.cluster.local:3178 ..."
```


so that pods can connect to it as described in https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#pod-identity

It does seem strange/brittle that I need to hardcode the namespace (`default`) here though.

Reply:
Do you really need the fully qualified domain here? In https://github.com/sourcegraph/sourcegraph/pull/12418/files seems to imply that the `gitserver-0.gitserver` part should be sufficient. It is also worth trying to see if `gitserver-0` is sufficient.

Reply:
I think you're right in that I don't need the entire domain name.

![screen shot 2018-08-09 at 1 53 18 pm](https://user-images.githubusercontent.com/9022011/43925309-c8eb49dc-9bdb-11e8-8e7a-724124c8d5f5.png)

Just `gitserver-0` didn't work when I tried it earlier. I don't think there are individual services for each replica - just the governing one.